application/common/components/Emailer.php

```diff
@@ -776,21 +776,38 @@ public function sendPasswordExpiringEmails()
             'status' => 'starting',
         ];
 
-        $users = User::getActiveUnlockedUsers();
+        $users = \Yii::$app->getDb()->createCommand("SELECT u.*
```

Comment:
I'm pretty sure you could use a non-active query to do the same thing.
$query = new Query(); // Not ActiveQuery, because that is slower.
$users = $query
-> select("u.*")
->from("user u")
->join('password p'.... Sorry, not quite up to thinking it all out...
->one();

Also, this was a pretty function. Nice one liner. Now it is a grotesque mess. Extract this perhaps?

Reply:
To each his own. I don't think SQL is "grotesque". But you're right that the function grew in complexity.